--- vagalume-0.7.1.orig/src/vgl-main-window.c.others	2009-02-26 09:12:11.000000000 -0500
+++ vagalume-0.7.1.orig/src/vgl-main-window.c	2009-05-13 12:19:23.168774960 -0400
@@ -770,7 +770,7 @@
         GtkWidget *play, *stop, *skip, *separ1, *separ2;
         GtkWidget *stopafter, *love, *ban, *tag, *dorecomm, *addtopls, *dload;
         GtkWidget *library, *neigh, *loved, *playlist, *recomm, *usertag;
-        GtkWidget *library2, *neigh2, *loved2, *playlist2;
+        GtkWidget *library2, *neigh2, *loved2, *playlist2, *recomm2, *usertag2;
         GtkWidget *managebmk, *bmkartist, *bmktrack;
         GtkWidget *about;
 #ifdef USE_HILDON_WINDOW
@@ -863,10 +863,14 @@
         neigh2 = gtk_menu_item_new_with_label(_("Neighbours..."));
         loved2 = gtk_menu_item_new_with_label(_("Loved tracks..."));
         playlist2 = gtk_menu_item_new_with_label(_("Playlist..."));
+        recomm2 = gtk_menu_item_new_with_label(_("Recommendations..."));
+        usertag2 = gtk_menu_item_new_with_label(_("Music tagged..."));
         gtk_menu_shell_append(othersub, library2);
         gtk_menu_shell_append(othersub, neigh2);
         gtk_menu_shell_append(othersub, loved2);
         gtk_menu_shell_append(othersub, playlist2);
+        gtk_menu_shell_append(othersub, recomm2);
+        gtk_menu_shell_append(othersub, usertag2);
         g_signal_connect(G_OBJECT(library2), "activate",
                          G_CALLBACK(others_radio_selected),
                          GINT_TO_POINTER(LASTFM_LIBRARY_RADIO));
@@ -879,6 +883,12 @@
         g_signal_connect(G_OBJECT(playlist2), "activate",
                          G_CALLBACK(others_radio_selected),
                          GINT_TO_POINTER(LASTFM_USERPLAYLIST_RADIO));
+        g_signal_connect(G_OBJECT(recomm2), "activate",
+                         G_CALLBACK(others_radio_selected),
+                         GINT_TO_POINTER(LASTFM_RECOMMENDED_RADIO));
+        g_signal_connect(G_OBJECT(usertag2), "activate",
+                         G_CALLBACK(others_radio_selected),
+                         GINT_TO_POINTER(LASTFM_USERTAG_RADIO));
 
         /* Actions */
         actions = GTK_MENU_ITEM(gtk_menu_item_new_with_mnemonic(_("_Actions")));
--- vagalume-0.7.1.orig/src/controller.c.other	2009-02-26 09:07:31.000000000 -0500
+++ vagalume-0.7.1.orig/src/controller.c	2009-05-13 12:31:12.382534553 -0400
@@ -1448,13 +1448,34 @@
                                                _("Play this user's radio"),
                                                friends,
                                                previous);
-        if (user != NULL) {
+	if (user == NULL) return;
+
+	/* Store the new value for later use */
+	g_free(previous);
+	previous = user;
+
+        if (type == LASTFM_RECOMMENDED_RADIO) {
+                url = lastfm_recommended_radio_url(
+                        user, 100);
+        } else if (type == LASTFM_USERTAG_RADIO) {
+                static char *previoustag = NULL;
+                char *tag;
+                tag = ui_input_dialog_with_list(
+                        vgl_main_window_get_window(mainwin, TRUE),
+                        _("Enter tag"), _("Enter one of your tags"),
+                        usertags, previoustag);
+                if (tag != NULL) {
+                        url = lastfm_usertag_radio_url(user, tag);
+                        /* Store the new value for later use */
+                        g_free(previoustag);
+                        previoustag = tag;
+                }
+        } else {
                 url = lastfm_radio_url(type, user);
+        }
+        if (url != NULL) {
                 controller_play_radio_by_url(url);
                 g_free(url);
-                /* Store the new value for later use */
-                g_free(previous);
-                previous = user;
         }
 }
 
